cfme/common/vm.py

```diff
@@ -192,6 +192,16 @@ def ip_address(self):
         """
         return find_pingable(self.mgmt)
 
+    @property
+    def all_ip_addresses(self):
```

Comment:
**Question** There is a method for each wrapanapi system called `all_ips`, I do see it seems to be limited on some providers. I think it be worth updating those methods if possible. A `TODO` here can be acceptable for now

Reply:
There is a bigger issue festering in this module I think, and that is all of these convenience methods/properties that are fetching data from the UI after navigation, and are not implemented in a contextual way.

I don't think switching them to wrapanapi queries directly is correct, as an existing TODO in the module suggests.

This is a place where, if we're going to have convenience methods for these meta values associated with an entity, they should be implemented using Sentaku in order to provide a consistent and predictable context for all of the values.  It will be too easy for their implementations to skew, and some to be using UI while others are using REST, or providers directly.

Provider direct queries should likely be a context of their own for Sentaku, that I don't believe we actually have yet. Separating `ViaRest` from `ViaMGMT`

Reply:
Hey @mshriver, I'm not sure if I understand you correctly, but are you suggesting we define all these convenient methods as contextual methods and properties?
`all_ips` is defined for almost all the providers in wrapanapi, but some of them return data and some don't.

Reply:
Hi @mshriver @jawatts I've currently added a TODO, because adding a new implementation and implementing sentaku will be a slightly bigger change. Can we please continue with a TODO for now?